### PR TITLE
Upgrade CPK/web3-react to finally get Safe WalletConnect detection

### PR DESCRIPTION
**Pull request by cag**
Based off of #1317 

Putting this here to get a Netlify deployment to try.

### CODE REVIEW

**Review by KadenZipfel — APPROVED** [review]

**Review by liaojikunwork — APPROVED** [review]

**Comment by cag** [comment]
This has now just become a big refactor, so yeah sorry, this is now a completely different animal.

**Review by KadenZipfel — APPROVED** [review]

File: app/src/components/market/sections/market_profile/market_status/open.tsx [hunk 1]
```diff
@@ -206,21 +206,19 @@ const Wrapper = (props: Props) => {
...
+                  buttonType={ButtonType.secondaryLine}
+                  onClick={() => {
+                    history.goBack()
+                  }}
+                >
+                  Back
+                </Button>
+                {isQuestionOpen ? openInRealitioButton : buySellButtons}
+              </StyledButtonContainer>
+            </WhenConnected>
```
Comment (KadenZipfel) [inline app/src/components/market/sections/market_profile/market_status/open.tsx]: I mistakenly made this change, should be the opposite, oops

**Review by hexyls — APPROVED** [review]
Awesome!

Just a note for us in the future, wallet connect will be disabled on Rinkeby after this PR is merged - we can fix that when [this issue](https://github.com/NoahZinsmeister/web3-react/issues/119 ) is resolved in web3-react.

File: app/src/hooks/connectedWeb3.tsx [hunk 2]
```diff
@@ -1,89 +0,0 @@
...
- * `useConnectedWeb3Context` safely to get web3 stuff without having to null check it.
- */
-export const ConnectedWeb3: React.FC = props => {
-  const [networkId, setNetworkId] = useState<number | null>(null)
-  const context = useWeb3Context()
-  const { account, active, error, library } = context
-
-  useEffect(() => {
-    let isSubscribed = true
-    const connector = localStorage.getItem('CONNECTOR')
```
Comment (hexyls) [inline app/src/hooks/connectedWeb3.tsx:43]: I noticed that one thing that we lose with the deletion of this file is auto login / auto connector detection on refresh.

I'm also using this file to [auto login safes](https://github.com/protofire/omen-exchange/pull/1317/commits/f411ddd13626eac89074afee69ecd37053b73c58#diff-1c8e03f884839b8a2150732e6e176df8c8fae34038fef297837bae585c6bd95cR53) in https://github.com/protofire/omen-exchange/pull/1317

Is there anyway we can do the upgrade without deleting this?

**Review by liaojikunwork — APPROVED** [review]

File: app/src/components/common/layout/footer/index.tsx [hunk 3]
```diff
@@ -40,14 +39,12 @@ const Break = styled.span`
...
 export const Footer = () => {
-  const context = useConnectedWeb3Context()
-  const { marketMakerFactory } = useContracts(context)
   return SHOW_FOOTER ? (
     <>
       <Wrapper paddingBottomSmall={DISCLAIMER_TEXT ? true : false}>
         <Link href="https://github.com/protofire/gnosis-conditional-exchange">Version {appVersion}</Link>
         <Break>·</Break>
-        <Link href={`https://etherscan.io/address/${marketMakerFactory.address}`}>Omen Contract</Link>
+        <Link href="https://etherscan.io/address/0x89023DEb1d9a9a62fF3A5ca8F23Be8d87A576220">Omen Contract</Link>
```
Comment (KadenZipfel) [inline app/src/components/common/layout/footer/index.tsx:47]: The contract address should be in the `constants` file

**Review by KadenZipfel — CHANGES_REQUESTED** [review]

File: app/src/components/common/layout/footer/index.tsx (hunk 3 above)
Comment (cag) [reply app/src/components/common/layout/footer/index.tsx]: Added it as `MARKET_MAKER_FACTORY_ADDRESS`

**Pull request closed without merging**